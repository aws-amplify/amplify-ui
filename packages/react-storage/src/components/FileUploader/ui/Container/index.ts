export { Container, ContainerProps } from './Container';
