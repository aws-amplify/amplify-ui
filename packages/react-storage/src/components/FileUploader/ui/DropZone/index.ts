export { DropZone } from './DropZone';
export { DropZoneProps } from './types';
