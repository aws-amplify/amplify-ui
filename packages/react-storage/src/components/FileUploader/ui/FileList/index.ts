export { FileList } from './FileList';
export { FileListProps } from './types';
