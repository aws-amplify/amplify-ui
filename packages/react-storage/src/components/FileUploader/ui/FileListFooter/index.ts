export type { FileListFooterProps } from './FileListFooter';
export { FileListFooter } from './FileListFooter';
