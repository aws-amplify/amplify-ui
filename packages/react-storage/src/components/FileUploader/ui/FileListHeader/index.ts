export { FileListHeader, FileListHeaderProps } from './FileListHeader';
