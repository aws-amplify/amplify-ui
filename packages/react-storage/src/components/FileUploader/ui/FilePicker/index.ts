export { FilePicker, FilePickerProps } from './FilePicker';
