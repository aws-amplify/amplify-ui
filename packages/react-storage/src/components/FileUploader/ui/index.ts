export { Container, ContainerProps } from './Container';
export { DropZone, DropZoneProps } from './DropZone';
export { FileList, FileListProps } from './FileList';
export { FileListHeader, FileListHeaderProps } from './FileListHeader';
export { FileListFooter, FileListFooterProps } from './FileListFooter';
export { FilePicker, FilePickerProps } from './FilePicker';
