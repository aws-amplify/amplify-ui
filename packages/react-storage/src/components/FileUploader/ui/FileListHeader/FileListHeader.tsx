import React from 'react';
import { FileUploaderDisplayTextDefault } from '../../utils';
import { ComponentClassName } from '@aws-amplify/ui';
import { Text } from '@aws-amplify/ui-react';

export interface FileListHeaderProps {
  allUploadsSuccessful: boolean;
  displayText: FileUploaderDisplayTextDefault;
  fileCount: number;
  remainingFilesCount: number;
  selectedFilesCount?: number;
}

export function FileListHeader({
  allUploadsSuccessful,
  displayText,
  fileCount,
  remainingFilesCount,
  selectedFilesCount = 0,
}: FileListHeaderProps): JSX.Element {
  const { getFilesUploadedText, getRemainingFilesText, getSelectedFilesText } =
    displayText;

  return (
    <Text className={ComponentClassName.FileUploaderPreviewerText}>
      {selectedFilesCount
        ? getSelectedFilesText(selectedFilesCount)
        : allUploadsSuccessful
        ? getFilesUploadedText(fileCount)
        : getRemainingFilesText(remainingFilesCount)}
    </Text>
  );
}
