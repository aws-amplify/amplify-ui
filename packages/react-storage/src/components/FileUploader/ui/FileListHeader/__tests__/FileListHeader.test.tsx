import * as React from 'react';
import { render } from '@testing-library/react';

import { ComponentClassName } from '@aws-amplify/ui';

import { FileListHeader, FileListHeaderProps } from '../FileListHeader';
import { defaultFileUploaderDisplayText } from '../../../utils/displayText';

const headerProps: FileListHeaderProps = {
  fileCount: 2,
  remainingFilesCount: 1,
  displayText: defaultFileUploaderDisplayText,
  allUploadsSuccessful: false,
};

describe('FileListHeader', () => {
  it('renders as expected when uploads are in progress', async () => {
    const { container, findByText } = render(
      <FileListHeader {...headerProps} />
    );

    expect(
      container.getElementsByClassName(
        `${ComponentClassName.FileUploaderPreviewerText}`
      )
    ).toHaveLength(1);

    expect(
      await findByText(
        headerProps.displayText.getRemainingFilesText(
          headerProps.remainingFilesCount
        )
      )
    ).toBeVisible();

    expect(container).toMatchSnapshot();
  });

  it('renders as expected when uploads are successful', async () => {
    const { container, findByText } = render(
      <FileListHeader {...headerProps} allUploadsSuccessful />
    );

    expect(
      await findByText(
        headerProps.displayText.getFilesUploadedText(headerProps.fileCount)
      )
    ).toBeVisible();

    expect(container).toMatchSnapshot();
  });
});
