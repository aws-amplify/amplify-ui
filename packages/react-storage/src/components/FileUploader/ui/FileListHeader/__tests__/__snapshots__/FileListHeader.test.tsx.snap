// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileListHeader renders as expected when uploads are in progress 1`] = `
<div>
  <p
    class="amplify-text amplify-fileuploader__previewer__text"
  >
    1 file uploading
  </p>
</div>
`;

exports[`FileListHeader renders as expected when uploads are successful 1`] = `
<div>
  <p
    class="amplify-text amplify-fileuploader__previewer__text"
  >
    2 files uploaded
  </p>
</div>
`;
